### PR TITLE
Rf leakage example

### PR DESCRIPTION
**Pull request by svandenb-dev**

### CODE REVIEW

File: pyaedt/hfss3dlayout.py [hunk 1]
```diff
@@ -145,6 +145,46 @@ def create_edge_port(self, primivitivename, edgenumber, iscircuit=True):
...
+
+        Parameters
+        ----------
+        primivitivenames : list(str)
+            List of the of the primitive name to create the wave port on.
+            The list length must be 2 e.g. for the two conductors or the command will not be executed.
+
+        edgenumbers :
+            List of the edge number to create the wave port on.
+            The list length must be 2 e.g. for thr two edges or the command will not be executed,
```
Comment (MaxJPRey) [inline pyaedt/hfss3dlayout.py:160]:
```suggestion
            The list length must be 2 e.g. for the two edges or the command will not be executed.
```

File: pyaedt/hfss3dlayout.py [hunk 2]
```diff
@@ -145,6 +145,46 @@ def create_edge_port(self, primivitivename, edgenumber, iscircuit=True):
         else:
             return False
 
+    @aedt_exception_handler
+    def create_wave_port_from_two_conductors(self, primivitivenames=[""], edgenumbers=[""]):
```
Comment (MaxJPRey) [inline pyaedt/hfss3dlayout.py:149]:
```suggestion
    def create_wave_port_from_two_conductors(self, primivitive_names=[""], edge_numbers=[""]):
```

File: pyaedt/hfss3dlayout.py [hunk 3]
```diff
@@ -145,6 +145,46 @@ def create_edge_port(self, primivitivename, edgenumber, iscircuit=True):
...
+        """Create a wave port.
+
+        Parameters
+        ----------
+        primivitivenames : list(str)
```
Comment (MaxJPRey) [inline pyaedt/hfss3dlayout.py:154]:
```suggestion
        primivitive_names : list(str)
```

File: pyaedt/hfss3dlayout.py [hunk 4]
```diff
@@ -145,6 +145,46 @@ def create_edge_port(self, primivitivename, edgenumber, iscircuit=True):
...
+            List of the of the primitive name to create the wave port on.
+            The list length must be 2 e.g. for the two conductors or the command will not be executed.
+
+        edgenumbers :
```
Comment (MaxJPRey) [inline pyaedt/hfss3dlayout.py:158]:
```suggestion
        edge_numbers :
```

File: pyaedt/hfss3dlayout.py [hunk 5]
```diff
@@ -145,6 +145,46 @@ def create_edge_port(self, primivitivename, edgenumber, iscircuit=True):
...
+            List of the edge number to create the wave port on.
+            The list length must be 2 e.g. for thr two edges or the command will not be executed,
+
+        Returns
+        -------
+        type
+            Name of the port when successful, ``False`` when failed.
+
+        """
+        if len(primivitivenames) == 2 and len(edgenumbers) == 2:
```
Comment (MaxJPRey) [inline pyaedt/hfss3dlayout.py:168]:
```suggestion
        if len(primivitive_names) == 2 and len(edge_numbers) == 2:
```

File: pyaedt/hfss3dlayout.py [hunk 6]
```diff
@@ -145,6 +145,46 @@ def create_edge_port(self, primivitivename, edgenumber, iscircuit=True):
...
+            listp = self.port_list
+            self.modeler.oeditor.CreateEdgePort(
+                [
+                    "NAME:Contents",
+                    "edge:="	, [			"et:="			, "pe",			"prim:="		, primivitivenames[0],			"edge:="		, edgenumbers[0]],
+                    "edge:="		, [			"et:="			, "pe",			"prim:="		, primivitivenames[1],			"edge:="		, edgenumbers[1]],
```
Comment (MaxJPRey) [inline pyaedt/hfss3dlayout.py:174]:
```suggestion
                    "edge:="	, [			"et:="			, "pe",			"prim:="		, primivitive_names[0],			"edge:="		, edge_numbers[0]],
                    "edge:="		, [			"et:="			, "pe",			"prim:="		, primivitive_names[1],			"edge:="		, edge_numbers[1]],
```

File: pyaedt/hfss3dlayout.py [hunk 7]
```diff
@@ -145,6 +145,46 @@ def create_edge_port(self, primivitivename, edgenumber, iscircuit=True):
...
+                    "external:="		, True,
+                    "btype:="		, 0
+                ]
+            )
+            listnew = self.port_list
+            a = [i for i in listnew if i not in listp]
```
Comment (MaxJPRey) [inline pyaedt/hfss3dlayout.py:180]:
```suggestion
            a = [i for i in self.port_list if i not in listp]
```

**Comment by MaxJPRey** [comment]
@svandenb-dev One of my main suggestion was to use the snake_case convention for the parameters name.

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 8]
```diff
@@ -0,0 +1,268 @@
+"""
+Example of fully parameterized design using edb.
+--------------------------------------
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:3]:
```suggestion
-------------------------------------------------------
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+    edb.core_stackup.stackup_layers.add_layer("bottom")
+    edb.core_stackup.stackup_layers.add_layer("dielectric", "bottom", layerType=1, thickness="275um",
+                                              material="FR4_epoxy")
+    edb.core_stackup.stackup_layers.add_layer("sig2", "dielectric")
+    edb.core_stackup.stackup_layers.add_layer("Diel_2", "sig2", layerType=1, thickness="275um", material="FR4_epoxy")
+    edb.core_stackup.stackup_layers.add_layer("sig1", "Diel_2")
+    edb.core_stackup.stackup_layers.add_layer("Diel_1", "sig1", layerType=1, thickness="275um", material="FR4_epoxy")
+    edb.core_stackup.stackup_layers.add_layer("top", "Diel_1")
+
+# Desgin variables #####
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:118]:
```suggestion
####################
# Design variables 
# 
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+edb.add_design_variable("$via_diam", "0.3mm")
+edb.add_design_variable("$pad_diam", "0.6mm")
+edb.add_design_variable("$anti_pad_diam", "0.7mm")
+edb.add_design_variable("$pcb_len", "30mm")
+edb.add_design_variable("$pcb_w", "5mm")
+edb.add_design_variable("$x_size", "1.2mm")
+edb.add_design_variable("$y_size", "1mm")
+edb.add_design_variable("$corner_rad", "0.5mm")
+
+# Via definition ####
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:132]:
```suggestion
#################
# Via definition 
# 
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 11]
```diff
@@ -0,0 +1,268 @@
...
+viadef1 = via_def(name="automated_via", hole_diam="$via_diam", pad_diam="$pad_diam", antipad_shape="Bullet",
+                  x_size="$x_size", y_size="$y_size", corner_rad="$corner_rad")
+viadef1.add_via_def_to_edb()
+
+# line creation ####
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:137]:
```suggestion
###################
# line creation 
# 
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 12]
```diff
@@ -0,0 +1,268 @@
...
+net_name = "strip_line_p"
+net_name2 = "strip_line_n"
+
+# line p ##################
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:141]:
```suggestion
######################
# line p 
# 
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+
+seg1_p.point_list = [["0.0", "($line_width+$line_spacing)/2"],
+                     ["$pcb_len/3-2*$via_spacing", "($line_width+$line_spacing)/2"],
+                     ["$pcb_len/3-$via_spacing", "($line_width+$line_spacing+$via_spacing)/2"],
+                     ["$pcb_len/3", "($line_width+$line_spacing+$via_spacing)/2"]
+                     ]
+seg1_p.place_line()
+path_port_1p = edb.core_primitives.primitives[-1]
+
+# via placement ####
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:152]:
```suggestion
#################
# Via placement 
# 
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 14]
```diff
@@ -0,0 +1,268 @@
...
+via_instance(pos_x="$pcb_len/3", pos_y="($line_width+$line_spacing+$via_spacing)/2", rotation=90,
+             net_name=net_name).place_via(viadef1)
+
+# line creation ####
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:156]:
```suggestion
##################
# line creation 
# 
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 15]
```diff
@@ -0,0 +1,268 @@
...
+seg2_p.point_list = [["$pcb_len/3", "($line_width+$line_spacing+$via_spacing)/2"],
+                     ["$pcb_len/3+$via_spacing", "($line_width+$line_spacing+$via_spacing)/2"],
+                     ["$pcb_len/3+2*$via_spacing", "($line_width+$line_spacing)/2"],
+                     ["2*$pcb_len/3-2*$via_spacing", "($line_width+$line_spacing)/2"],
+                     ["2*$pcb_len/3-$via_spacing", "($line_width+$line_spacing+$via_spacing)/2"],
+                     ["2*$pcb_len/3", "($line_width+$line_spacing+$via_spacing)/2"]
+                     ]
+seg2_p.place_line()
+
+# via placement ####
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:167]:
```suggestion
##################
# Via placement
# 
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 16]
```diff
@@ -0,0 +1,268 @@
...
+via_instance(pos_x="2*$pcb_len/3", pos_y="($line_width+$line_spacing+$via_spacing)/2", rotation=90,
+             net_name=net_name).place_via(viadef1)
+
+# line creation ####
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:171]:
```suggestion
#################
# Line creation
# 
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 17]
```diff
@@ -0,0 +1,268 @@
...
+seg3_p = line(width="$line_width", net_name=net_name, layer="top")
+seg3_p.point_list = [["2*$pcb_len/3", "($line_width+$line_spacing+$via_spacing)/2"],
+                     ["2*$pcb_len/3+$via_spacing", "($line_width+$line_spacing+$via_spacing)/2"],
+                     ["2*$pcb_len/3+2*$via_spacing", "($line_width+$line_spacing)/2"],
+                     ["$pcb_len", "($line_width+$line_spacing)/2"]
+                     ]
+seg3_p.place_line()
+path_port_2p = edb.core_primitives.primitives[-1]
+
+# line n ##################
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:181]:
```suggestion
##################
# line n
# 
```

File: examples/00-basic-examples/04_edb_parametrized_design.py [hunk 18]
```diff
@@ -0,0 +1,268 @@
...
+
+# line creation ####
```
Comment (maxcapodi78) [inline examples/00-basic-examples/04_edb_parametrized_design.py:183]:
```suggestion
################
# line creation
# 
```

**Review by maxcapodi78 — APPROVED** [review]
Please modify the examples comments like suggested

**Pull request merged**